### PR TITLE
LocateAnything 3B

### PR DESCRIPTION
**Pull request by SangbumChoi**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=46300)

# What does this PR do?

Adds native Transformers support for NVIDIA LocateAnything 3B, including configuration, image processor, processor, PyTorch modeling code, AutoClass registration, documentation, and focused tests.

This is currently draft/WIP because the required maintainer or issue coordination has not been linked yet, and several source files still carry restrictive NVIDIA license headers that need a clean licensing path before review.

## Coordination and duplicate-work status

- Coordination/approval issue or forum link: missing. This must be added before requesting review.
- Duplicate PR search run: `gh pr list --repo huggingface/transformers --state open --search "LocateAnything OR Locate Anything OR nvidia/LocateAnything" --limit 20`. The only direct LocateAnything PR found was this PR (#46300).
- Issue search run: `gh issue list --repo huggingface/transformers --state open --search "LocateAnything OR Locate Anything OR nvidia/LocateAnything" --limit 20`. No matching open issue was found.

## Tests and checks run

- `make style`
- `make typing`
- `PYTHONPATH=src python utils/check_repo.py`
- `make check-repo`
- `PYTHONPATH=src pytest tests/models/locateanything/test_modeling_locateanything.py tests/models/locateanything/test_image_processing_locateanything.py -q`
- `PYTHONPATH=src python - <<'PY'
from transformers import AutoProcessor
processor = AutoProcessor.from_pretrained("nvidia/LocateAnything-3B", trust_remote_code=False)
print(type(processor).__name__)
print(type(processor.image_processor).__name__)
PY`

## AI assistance disclosure

AI assistance was used to draft and iterate on this patch. The human submitter is responsible for reviewing every changed line and validating the change end-to-end before moving this PR out of draft.

## Before submitting

- [ ] This is not a pure code-agent PR.
- [ ] I have read the contributor guidelines.
- [ ] This was discussed/approved via a GitHub issue or forum thread, linked above.
- [x] Documentation was added/updated.
- [x] Tests were added/updated.

### CODE REVIEW

**Comment by SangbumChoi** [comment]
@molbap @merveenoyan soft ping if you have time to review

**Comment by IMvision12** [comment]
@SangbumChoi it will be good to have a colab gist with inference

**Comment by molbap** [comment]
@IMvision12 Or a snippet is already good, plain copy-able snippets are great for verifiability! 
@SangbumChoi see also my comment on the other PR. Not much bandwidth on either right now

**Comment by SangbumChoi** [comment]
Thanks, here is a plain Colab-copyable native inference snippet. It is pinned to the current PR head and model revision, and does not use `trust_remote_code`.

```python
%pip install -q "git+https://github.com/SangbumChoi/transformers.git@0aff35412bdea44d9c140a6b2d4728f3857d1fb6"

import requests
import torch
from PIL import Image
from transformers import AutoProcessor, LocateAnythingForConditionalGeneration

assert torch.cuda.is_available(), "Select a GPU runtime first."

model_id = "nvidia/LocateAnything-3B"
model_revision = "c32291ca5e996f5a7a485845b4f57a233936bba0"
dtype = torch.bfloat16 if torch.cuda.is_bf16_supported() else torch.float16

processor = AutoProcessor.from_pretrained(model_id, revision=model_revision)
model = LocateAnythingForConditionalGeneration.from_pretrained(
    model_id,
    revision=model_revision,
    dtype=dtype,
    attn_implementation="sdpa",
).to("cuda").eval()

response = requests.get(
    "https://images.cocodataset.org/val2017/000000039769.jpg",
    stream=True,
    timeout=30,
)
response.raise_for_status()
image = Image.open(response.raw).convert("RGB")

question = "Locate all the instances that matches the following description: cat."
messages = [
    {
        "role": "user",
        "content": [
            {"type": "image", "image": image},
            {"type": "text", "text": question},
        ],
    }
]
prompt = processor.apply_chat_template(messages, add_generation_prompt=True, tokenize=False)
inputs = processor(images=image, text=[prompt], return_tensors="pt").to("cuda")

with torch.inference_mode():
    generated = model.generate(**inputs, do_sample=False, max_new_tokens=64)

answer = processor.decode(
    generated[0, inputs["input_ids"].shape[-1] :],
    skip_special_tokens=False,
)
print(answer)
```

Expected decoded output:

```text
<ref>cat</ref><box><0><115><494><988></box><|im_end|>
```

**Comment by SangbumChoi** [comment]
I also uploaded a native Transformers training example without changing this PR branch:

- [LoRA training script](https://github.com/SangbumChoi/transformers/blob/codex/locateanything-training-demo/examples/pytorch/locateanything/locateanything_lora_train.py)
- [README with Colab/GPU commands](https://github.com/SangbumChoi/transformers/blob/codex/locateanything-training-demo/examples/pytorch/locateanything/README.md)

```python
%pip install -q \
  "git+https://github.com/SangbumChoi/transformers.git@38b9bd8d87cd52ffca787490de353004a71de1c1" \
  "peft>=0.17.0" requests pillow

!wget -q \
  https://raw.githubusercontent.com/SangbumChoi/transformers/codex/locateanything-training-demo/examples/pytorch/locateanything/locateanything_lora_train.py

# Processor and assistant-only label validation (no model weights):
!python locateanything_lora_train.py --dry-run

# One LoRA optimizer step and adapter save:
!python locateanything_lora_train.py --steps 1 --output-dir locateanything-warehouse-lora
```

The default is the harder warehouse/industrial integration case (forklift, pallets, stacked boxes, and aisle). It limits preprocessing to 8,192 vision patches, yielding 2,226 total tokens and 62 supervised assistant tokens. I validated exact prompt-prefix masking and target decoding, plus a tiny LocateAnything LoRA forward/backward pass with finite loss and gradients on all trainable adapter tensors; `make style` also passes.

Scope: this exercises standard autoregressive NTP SFT in native Transformers. It does **not** reproduce NVIDIA's custom PBD/MTP continual-SFT stack; the full pipeline remains in the [official Eagle training guide](https://github.com/NVlabs/Eagle/blob/main/Embodied/document/TRAINING.md). The full 3B optimizer step has not been run from this environment because Hugging Face Jobs is not authenticated here, so an A10G/L4/A100/H100 GPU run is still required for end-to-end verification.

**Comment by SangbumChoi** [comment]
@IMvision12 Would you willing to review the code?
